subsys/event_manager/event_manager.c

```diff
@@ -116,6 +116,16 @@ void __weak event_manager_trace_event_submission(const struct event_header *eh,
 {
 }
 
+void * __weak event_manager_alloc(size_t size)
```

Comment:
I think we should move error handling here to allow user to customize it. That would give user flexibility in OOM error handling.

Reply:
The error handling code would not be duplicated too

Reply:
This was actually one of the discussion points on the RFC :/